s3_transport/include/irods/private/s3_transport/s3_transport.hpp

```diff
@@ -1478,14 +1478,20 @@ namespace irods::experimental::io::s3_transport
                         // turn off trunc flag
                         std::ios_base::openmode mode;
                         bool trunc_flag = false;
-                        if (data.threads_remaining_to_close == 0) {
-                            trunc_flag = true;
+
+						// If we know the number of threads, use threads_remaining_to_close to determine if this is the
+						// first open. If we do not know the number of threads, use file_open_counter.
```

Comment:
The explanatory comment above says to do this or that based on whether this is the "first open", so I'm wondering why we wouldn't always use `file_open_counter` as our deciding metric. Doesn't `file_open_counter` reveal whether this is the first open all by itself, regardless of the number of threads or whether or not we know how many threads there are?

This may be a bigger question which we can talk about later, so don't let this hold this up. This works, so, it's more important that we ship the fix, IMO.